fastjet/internal/delaunay/triangle.go

```diff
@@ -0,0 +1,206 @@
+// Copyright 2017 The go-hep Authors.  All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package delaunay
+
+import (
+	"fmt"
+	"math"
+)
+
+// three points that make up the triangle
+// plus hierarchical information to find triangles
+type Triangle struct {
+	children []*Triangle // Triangles that resulted from changing up this triangle
+	A        *Point
+	B        *Point
+	C        *Point
+}
+
+func NewTriangle(a *Point, b *Point, c *Point) *Triangle {
+	// make it counter clockwise
+	if !cClockwise(a, b, c) {
+		b, c = c, b
+	}
+	t := &Triangle{
+		A: a,
+		B: b,
+		C: c,
+	}
+	t.children = make([]*Triangle, 0, 10)
+	return t
+}
+
+func cClockwise(a *Point, b *Point, c *Point) bool {
+	return (b.Y-a.Y)*(c.X-b.X)-(c.Y-b.Y)*(b.X-a.X) < 0
+}
+
+// checking if the point is in the triangle and if it is on an edge
+// using barycentric coordinates
+func (t *Triangle) inTriangle(p *Point) (bool, bool) {
+	barcen1 := ((t.B.Y-t.C.Y)*(p.X-t.C.X) + (t.C.X-t.B.X)*(p.Y-t.C.Y)) / det(t)
+	barcen2 := ((t.C.Y-t.A.Y)*(p.X-t.C.X) + (t.A.X-t.C.X)*(p.Y-t.C.Y)) / det(t)
+	barcen3 := 1 - barcen1 - barcen2
+
+	// inside triangle
+	if barcen1 > 0 && barcen1 < 1 && barcen2 > 0 && barcen2 < 1 && barcen3 > 0 && barcen3 < 1 {
+		return true, false
+	}
+	// outside triangle
+	if barcen1 < 0 || barcen2 < 0 || barcen3 < 0 {
+		return false, false
+	}
+	// otherwise on the edge
+	return true, true
+}
+
+func det(t *Triangle) float64 {
+	return (t.B.Y-t.C.Y)*(t.A.X-t.C.X) + (t.C.X-t.B.X)*(t.A.Y-t.C.Y)
+}
+
+// return true if the point is inside the circumcircle of the triangle
```

Comment:
"// inCircumCircle returns whether the point is inside the circumcircle of the triangle."

and rename the method.